FBAllocationTracker/Generations/FBAllocationTrackerGeneration.mm

```diff
@@ -45,12 +47,22 @@
         /**
          Retain object and add it to returnValue.
          This operation can be unsafe since we are retaining object that could
-         be deallocated on other thread.
+         be deallocated on other thread. 
+         
+         When NSZombie enabled, we can find if object has been deallocated by checking its class name.
          */
-        returnValue.push_back(object);
+        if (fb_isNSZombieEnabled()) {
+          if (!fb_isZombieObject(object)) {
```

Comment:
I think it might be fine to make only on if call here, if you do "isZombieObject", it is sort of implied that you need zombies enabled for that. Thoughts?